### DISCUSSION TITLE
Logical Thinking in Programming

### DISCUSSION THREAD

**Discussion by Blacktrecks** [community/community#38375] [Other Feature Feedback, Questions, & Ideas]
Fast ways to develop that programming thinking?
or slow, I start to understand but I need more info.

**Comment by jfullstackdev** [comment]
here is the thing in actual development (in full-stack web development context),

you need CRUD, like MySQL, this will store data for the dynamic pages, so you need to develop that database skill

there is also the core of programming, where, for example, you are dealing a lot of Arrays, because it's essential, data structures, data types, etc., this is where you need to use a lot of logic and problem solving

and take note, they are really the same concepts no matter what programming language you are in, they differ in terms of optimization

and one of the challenging part I mostly encountered is to continue working on in a codebase that others started but is not maintainable in the long run, you should not repeat the same process or else you will not meet deadlines or worse be fired for being inefficient and too slow, so you need to think of really ingenious way how to make the existing code work

check this out

https://github.com/jdevstatic/programming-core-concepts

you need to understand also at least a full-stack web framework, like PHP / Laravel or Java / SpringBoot, it's demanding particularly for small to medium companies that do not have so much budgeting, well even top-tech companies do layoffs because of budgeting

and lastly, coding is a double-edge sword, most of the time it's very frustrating like you want to give up the same day, on the other hand, once you successfully do the tasks, it's very rewarding
